sprint-4/step-1

Neverhood93 opened the pull request:
Интерфейсы приведены в единый нейминг.

Neverhood93 committed:
Интерфейсы приведены в единый нейминг.

mikebars1995 commented on src/pages/login/loginPage.tsx:
```diff
@@ -20,7 +20,7 @@ import Preloader from "../../components/common/preloader/preloader";
+  const { formState, handleFieldChange } = useForm<ILoginRequest>({
     email: "",
     password: "",
   });
```
**Можно лучше**



Скрин https://disk.yandex.ru/i/IQRKDKxbsL_Qrg . Все события обработчиков нужно **правильно** типизировать. Это легко сделать. Для
`onChange` есть тип `ChangeEvent<HTMLInputElement>`, а для `onSubmit` есть
`FormEvent<HTMLFormElement>` в библиотеке `react`. Для обычных функций есть тип
`SyntheticEvent` для их событий. Тип `Response` для ответа от сервера, тип `KeyboardEvent`
для событий клавиатуры, тип `RouteProps` для рутов



**Можно лучше**
* Есть небольшая хитрость: можно всегда проверить, какой тип у функции или события.
Для этого нужно навести мышкой на атрибут `onChange`, `onSubmit`, `onClick` и тд. Там
высветится типизация того, что принимают эти атрибуты.

mikebars1995 reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Посмотрите на **гитхабе** все комментарии к коду (нужно прокрутить вниз страницу там)

Работа проделана огромная


Что сделано отлично:


* Отлично, что знаете про тип `Response` для ответа от сервера
* Отлично, что знаете про тип `KeyboardEvent` для событий клавиатуры



Что нужно исправить:

* Скрин https://disk.yandex.ru/i/IQRKDKxbsL_Qrg . Все события обработчиков нужно **правильно** типизировать. 

Что можно улучшить:

* Есть небольшая хитрость: можно всегда проверить, какой тип у функции или события.
Для этого нужно навести мышкой на атрибут `onChange`, `onSubmit`, `onClick` и тд. Там
высветится типизация того, что принимают эти атрибуты.

Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте работоспособность проекта и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

**Комментарии `Можно лучше` не обязательны к исправлению прямо сейчас, это рекомендации**

Удачного рефакторинга кода. 

Neverhood93 committed:
Исправление замечаний:
"Все события обработчиков нужно правильно типизировать." - изменено FormEvent на FormEvent<HTMLFormElement>.

mikebars1995 reviewed (commented):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
